api/users/views.py

```diff
@@ -784,14 +784,21 @@ def post(self, request, *args, **kwargs):
 
         if claimer.is_anonymous and email:
             claimer = get_user(email=email)
-            if claimer and claimer.is_registered:
-                self._send_claim_email(claimer, claimed_user, record_referent, registered=True)
-            else:
-                self._send_claim_email(email, claimed_user, record_referent, notify=True, registered=False)
+            try:
+                if claimer and claimer.is_registered:
+                    self._send_claim_email(claimer, claimed_user, record_referent, registered=True)
+                else:
+                    self._send_claim_email(email, claimed_user, record_referent, notify=True, registered=False)
+            except HTTPError as e:
+                raise ValidationError(e.data['message_long'])
         elif isinstance(claimer, OSFUser):
             if unclaimed_record.get('referrer_id', '') == claimer._id:
                 raise ValidationError('Referrer cannot claim user.')
-            self._send_claim_email(claimer, claimed_user, record_referent, registered=True)
+            try:
+                self._send_claim_email(claimer, claimed_user, record_referent, registered=True)
+            except HTTPError as e:
+                raise ValidationError(e.data['message_long'])
```

Comment:
@adlius, can you add a test for this behavior?

Reply:
Tried to add test for this. But somehow the added tests that make two consecutive API calls which results in two calls to `_send_claim_email()` errors with `InterfaceError: connection already closed`. Tried couple of things but none seems to work.

Reply:
@adlius, instead of sending two requests, you can fake the first one was sent by manually added 'last_sent' to the unclaimed records.  I'll go ahead and merge and add the test myself.

```
    def test_claim_auth_throttle_error(self, app, url, claimer, unreg_user, project, mock_mail):
        unreg_user.unclaimed_records[project._id]['last_sent'] = timezone.now()
        unreg_user.save()
        res = app.post_json_api(
            url.format(unreg_user._id),
            self.payload(id=project._id),
            auth=claimer.auth,
            expect_errors=True
        )
        assert res.status_code == 400
        assert res.json['errors'][0]['detail'] == 'User account can only be claimed with an existing user once every 24 hours'
        assert mock_mail.call_count == 0
```

Reply:
Thanks.